### PR TITLE
[Bug][API]edit workflow query resources list error.

### LINKED ISSUE #12573: [Bug][API]edit workflow  query resources list error.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

edit workflow error.

log:
ERROR] 2022-10-27 22:26:50.449 +0800 org.apache.dolphinscheduler.api.exceptions.ApiExceptionHandler:[53] - 查询资源列表错误
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl.queryResourceList(ResourcesServiceImpl.java:877)
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl$$FastClassBySpringCGLIB$$d2d5d7f6.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy.invokeMethod(CglibAopProxy.java:386)
	at org.springframework.aop.framework.CglibAopProxy.access$000(CglibAopProxy.java:85)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:704)
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl$$EnhancerBySpringCGLIB$$a71e6a3d.queryResourceList(<generated>)
	at org.apache.dolphinscheduler.api.controller.ResourcesController.queryResourceList(ResourcesController.java:209)
	at org.apache.dolphinscheduler.api.controller.ResourcesController$$FastClassBySpringCGLIB$$2270281e.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
	at org.apache.dolphinscheduler.api.aspect.AccessLogAspect.doAround(AccessLogAspect.java:105)
	at sun.reflect.GeneratedMethodAccessor228.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at org.apache.dolphinscheduler.api.controller.ResourcesController$$EnhancerBySpringCGLIB$$e73f5ab9.queryResourceList(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1070)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1656)
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626)
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:552)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:772)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:748)

### What you expected to happen

create workflow.

### How to reproduce

create workflow.
save workflow.
edit workflow.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hstdream**
## Purpose of the pull request

close #12573
## Brief change log

image

## Verify this pull request

edit workflow

### CODE REVIEW

**Comment by caishunfeng** [comment]
I found that this pr can not resolve the issue, when I want to add new shell task, it will query resources and throw error too.

I think we should change the front part, if fullName is empty, don't request resource list.
image

**Comment by hstdream** [comment]
> I found that this pr can not resolve the issue, when I want to add new shell task, it will query resources and throw error too.
> 
> I think we should change the front part, if fullName is empty, don't request resource list. image

Yours is another request, and reporting an error is not a problem with me

**Comment by caishunfeng** [comment]
> > I found that this pr can not resolve the issue, when I want to add new shell task, it will query resources and throw error too.
> > I think we should change the front part, if fullName is empty, don't request resource list. image
> 
> Yours is another request, and reporting an error is not a problem with me

related pr #12076   cc @SbloodyS @EricPyZhou

**Comment by EricPyZhou** [comment]
@hstdream please confirm two things:
1. Do you login as admin user?
2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
If not, then it will throw this error.

**Comment by calvinjiang** [comment]
> @hstdream please confirm two things:
> 
> 1. Do you login as admin user?
> 2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
>    If not, then it will throw this error.

@hstdream @caishunfeng I think one of the most important issues is the error message. That couldn't tell users what the exact problem is. The error message should tell users like 'You need to set the resource storage in the file of common.properties.'

**Comment by EricPyZhou** [comment]
> > @hstdream please confirm two things:
> > 
> > 1. Do you login as admin user?
> > 2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
> >    If not, then it will throw this error.
> 
> @hstdream @caishunfeng I think one of the most important issues is the error message. That couldn't tell users what the exact problem is. The error message should tell users like 'You need to set the resource storage in the file of common.properties.'



> > @hstdream please confirm two things:
> > 
> > 1. Do you login as admin user?
> > 2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
> >    If not, then it will throw this error.
> 
> @hstdream @caishunfeng I think one of the most important issues is the error message. That couldn't tell users what the exact problem is. The error message should tell users like 'You need to set the resource storage in the file of common.properties.'

#12901 now in this issue, we propose to return empty list when a tenant resource path is not found.

**Comment by davidzollo** [comment]
> > @hstdream please confirm two things:
> > 
> > 1. Do you login as admin user?
> > 2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
> >    If not, then it will throw this error.
> 
> @hstdream @caishunfeng I think one of the most important issues is the error message. That couldn't tell users what the exact problem is. The error message should tell users like 'You need to set the resource storage in the file of common.properties.'

+1, ` The error message should tell users like 'You need to set the resource storage in the file of common.properties.'  ` is a better suggestion  @hstdream what do you think?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java [hunk 1]
```diff
@@ -847,7 +847,12 @@ private boolean upload(User loginUser, String fullName, MultipartFile file, Reso
     @Override
     public Map<String, Object> queryResourceList(User loginUser, ResourceType type, String fullName) {
         Map<String, Object> result = new HashMap<>();
-
+        if (!PropertyUtils.getResUploadStartupState()) {
+            logger.error("Storage does not start up, resource upload startup state: {}.",
+                    PropertyUtils.getResUploadStartupState());
```
Comment (davidzollo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java:852]: The error message should tell users like 'You need to set the resource storage in the file of common.properties.'

**Comment by EricPyZhou** [comment]
> > > @hstdream please confirm two things:
> > > 
> > > 1. Do you login as admin user?
> > > 2. Do all tenantCodes used by all users have its path defined in your environment? For example, if you are using S3 bucket, verify that if `tenantCode/resources/` exist
> > >    If not, then it will throw this error.
> > 
> > 
> > @hstdream @caishunfeng I think one of the most important issues is the error message. That couldn't tell users what the exact problem is. The error message should tell users like 'You need to set the resource storage in the file of common.properties.'
> 
> +1, `The error message should tell users like 'You need to set the resource storage in the file of common.properties.' ` is a better suggestion @hstdream what do you think?

Please shift discussion into #12901 as it is the dedicated one for designing implementations on resolving this issue.

**Pull request closed without merging**